estmator_project/estmator_project/views.py

```diff
@@ -110,3 +127,165 @@ def client_list_form_view(request):
+        if 'travel_time' in request.POST and request.POST['travel_time'] != '':
+            quote.travel_time = request.POST['travel_time']
+        else:
+            quote.travel_time = 0
+
+        quote.org_street_load = 'org_street_load' in request.POST
+        quote.org_midrise_elev_std = 'org_midrise_elev_std' in request.POST
+        quote.org_midrise_elv_frt = 'org_midrise_elv_frt' in request.POST
+        quote.org_highrise = 'org_highrise' in request.POST
+        quote.org_stairs = 'org_stairs' in request.POST
+        quote.org_lng_psh = 'org_lng_psh' in request.POST
+
+        quote.dest_street_load = 'dest_street_load' in request.POST
+        quote.dest_midrise_elev_std = 'dest_midrise_elev_std' in request.POST
+        quote.dest_midrise_elv_frt = 'dest_midrise_elv_frt' in request.POST
+        quote.dest_highrise = 'dest_highrise' in request.POST
+        quote.dest_stairs = 'dest_stairs' in request.POST
+        quote.dest_lng_psh = 'dest_lng_psh' in request.POST
+
+        quote.save()
+        products = request.POST.getlist('product')
+        counts = [int(x) for x in request.POST.getlist('product_count')]
+
+        context['categories'] = {c: [] for c in Category.objects.all()}
+        for i, count in enumerate(counts):
+            if count > 0:
+                prop = ProductProperties()
+                prop.quote = quote
+                prop.product = Product.objects.get(id=int(products[i]))
+                prop.count = count
+                prop.save()
+
+                quote.productproperties_set.add(prop)
+                context['categories'][prop.product.category].append((prop.product, count))
+
+        quote.save()
+        context['quote'] = quote
+        context['straight_time_cost'] = request.POST['straight_time_cost']
+        context['over_time_cost'] = request.POST['over_time_cost']
+
+        """
+        Duplicating getting the categories/products/counts for the
+        context because I can't figure out what the above is doing?
+
+        Will try again when I get some sleep.
+        """
+        context['categories'] = None
+        quote_products = ProductProperties.objects.filter(quote=quote)
+        categories = {}
+
+        for c in Category.objects.all():
+            categories[c.name] = []
+            for qp in quote_products:
+                if qp.product.category == c:
+                    categories[c.name].append({
+                        'name': qp.product.name,
+                        'count': qp.count
+                    })
+        """
+        figure out better way to only include the needed categories
+        """
+        for c in categories.keys():
+            if categories[c] == []:
+                del categories[c]
+
+        context['categories'] = categories
+        context['st'] = float(quote.grand_total) / 60 * float(quote.client.company.st_rate)
+        context['ot'] = float(quote.grand_total) / 60 * float(quote.client.company.ot_rate)
```

Comment:
Pretty much this entire view feels to me like it should be given over to either api methods on the Quote class or a special form that does all these calculations for you.  This is a very long view containing business logic, and that's not a very Django thing to do.